dogfood/Dockerfile.nix

```diff
@@ -0,0 +1,49 @@
+# Build stage
+FROM nixos/nix:2.19.2 as nix
+
+# enable --experimental-features 'nix-command flakes' globally
```

Comment:
As a general rule, a comment should describe _why_ we're doing something, it's usually not as important to say _what_ we're doing (it's usually deducible from the code itself).

To put simply, a reader might still wonder why we need to enable this globally.